### ISSUE TITLE
Autologin through URL

### ISSUE THREAD

**Issue by alanmilinovic** [OliveTin/OliveTin#448]
Please consider adding login through URL by putting username and passwd. Something like -> http://IP:1337/Dashboard?username=XXXXX&passwd=YYYYY

Not sure if you have doubts regarding security.
Alternative is to support some kind of token in url, instead of plain text with username and password?

**Comment by jamesread** [comment]
Heya @alanmilinovic , what would be the use case for this?

**Comment by alanmilinovic** [comment]
Hey James, my OliveTin  is embedded into [Organizr](https://github.com/causefx/Organizr) another application where I already have authentication covered. Therefore URL based login can streamline and simplify my user experience by skipping the separate login screen. 

Another case would be sharing the URL with my family members that can use some of the actions without manually logging in.

I know that this could impose a question about security risk. In my case I do have https to prevent network interception.

**Comment by jamesread** [comment]
@alanmilinovic , I origianally wrote JWT auth support because I use Organizr myself; https://docs.olivetin.app/jwt-hmac.html

If I documented this for Organizer specifically a little better, would you be prepared to use Organizr's JWT auth support instead of this "magic URL" type login?

**Comment by alanmilinovic** [comment]
> @alanmilinovic , I originally wrote JWT auth support because I use Organizr myself; https://docs.olivetin.app/jwt-hmac.html
> 
> If I documented this for Organizer specifically a little better, would you be prepared to use Organizr's JWT auth support instead of this "magic URL" type login?

Yes, I can live with JWT for Organizr case but for the other case where I would like to share direct link with other people it will not work because they will not use Organizr.

**Comment by jamesread** [comment]
Okay, would these users already exist in the OliveTin config? If so, you could just use local users and copy people their username/password? 

Or, would static users, each which have a token for a "magic login link" be ok? 

The biggest thing I want to avoid is making OliveTin be able to dynamically write/add config options as it opens up a whole world of issues.

**Comment by alanmilinovic** [comment]
> Okay, would these users already exist in the OliveTin config? If so, you could just use local users and copy people their username/password?
> 
> Or, would static users, each which have a token for a "magic login link" be ok?
> 
> The biggest thing I want to avoid is making OliveTin be able to dynamically write/add config options as it opens up a whole world of issues.

Yes, they would already be defined in config.yaml file. There is no need to dynamically manipulate config options for sure.

**Comment by jamesread** [comment]
Okay, so the sending of the username and password via a URL, is simply so they don't have to type it into the login form? This feature would just save these users a few seconds of time typing?

**Comment by alanmilinovic** [comment]
> Okay, so the sending of the username and password via a URL, is simply so they don't have to type it into the login form? This feature would just save these users a few seconds of time typing?

Yes, I mean it is also some of them tend to forget their password so it is not just about a few seconds. But hey James this was only my wish, you don't need to implement it. :)

**Comment by jamesread** [comment]
Hey @alanmilinovic , OliveTin recently got support for "deep linking", where you could fill out argument values and have those included in the URL - allowing you to send a "pre filled" form to someone as a link. 

It would have been technically possible to add this for the User Login as well, but I'm uneasy about passing passwords in plaintext via a URL - while it could be useful for your usecase, I'm worried that it could created unintended problems or even security issues in the future for other users in the future.

I think about all the other apps I use, and I don't think there's a single app I have that has this behaviour. 

Sorry to disappoint, but I've decided not to implement this for the following reasons;

- Could cause possible security problems in the future for other users
- Unusual behaviour for any app
- No other users asking for this (yet!)

**Comment by alanmilinovic** [comment]
Hi @jamesread, thanks for getting back to me about this topic. I do understand your concerns, specially as I come from cyber security world.

And what about alternative solution like sending a token in URL that would not be the pure plain text, as I mentioned on my original post? If that is something you could think about potential solution, would be nice.
